### PR TITLE
feat: prompt user to install pwa

### LINKED ISSUE #533: Add a prompt modal asking to install PWA
Superteam Earn is now fully optimised for mobile. We want to educate and prompt our mobile users to install the Earn PWA.

- Show this modal to Mobile users
    image

- The modal should be shown once per user (use localstorage to track this), and should only be shown on mobile. Use user-agents to identify mobile device
- Check out [this guide](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Making_PWAs_installable) on how to make PWA installable in one click, from the mobile browser.  But would advice to do more research on the same, in order to make this work for both iOS and Android devices, as well as both Safari and Chromium browsers.

In browsers which the one-click installation is not possible, show the following modal:
image

[Design Link](https://www.figma.com/file/E4lVMRBOg3yDodMpdgdALn/ST-Earn%3A-Mobile-Redesign?type=design&node-id=117-237&mode=design&t=OTTR6XFG0LotHQm4-4) for both modals

### PR DESCRIPTION
**Pull request by SkTomsi**
## What does this PR do?
closes #533 
## Screenshots (if appropriate)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "next-auth": "^4.24.6",
     "next-pwa": "^5.6.0",
     "next-seo": "^6.5.0",
+    "next-useragent": "^2.8.0",
```
Comment (a20hek) [inline package.json:62]: i dont think a separate package is needed
from memory, i think android devices can be checked using something like
```
const isAndroid = (): boolean => {
  return /Android/i.test(navigator.userAgent);
};
```

there is something similar for iOS too

File: src/pages/index.tsx [hunk 2]
```diff
@@ -80,9 +93,45 @@ const HomePage: NextPage = () => {
...
+    };
+
+    setTimeout(() => {
+      showInstallAppModal();
+    }, 1000);
+  }, [userInfo]);
+
+  const installApp = async () => {
+    if (installPrompt.current) {
+      // @ts-expect-error --> to remove prompt doesnt exist error
```
Comment (a20hek) [inline src/pages/index.tsx:117]: we generally don't encourage `ts-ignore`s and `ts-expect-error`s
so please find a proper typesafe alternative

File: src/pages/index.tsx [hunk 3]
```diff
@@ -80,9 +93,45 @@ const HomePage: NextPage = () => {
     updateFeatureModalShown();
   }, [userInfo]);
 
+  useEffect(() => {
```
Comment (a20hek) [inline src/pages/index.tsx:96]: would be better to show the modal ~30 seconds into opening the site

**Review by a20hek — CHANGES_REQUESTED** [review]

**Review by a20hek — APPROVED** [review]
lgtm!

**Comment by a20hek** [comment]
just a heads up. there was just one issue with the pwa check; had to add a few more checks: https://github.com/SuperteamDAO/earn/commit/3cde4fab01a5f711d49bf7f60f43a09952787f0e#diff-18e0d4553c97cfc420e938ccafb4e3a688e782fa4512ba4ceae3e2a6f24c1987L112-R121

have fixed it; but just letting you know @SkTomsi

**Comment by SkTomsi** [comment]
ahhh i missed my bad, thanks for the addition @a20hek

**Pull request merged**